arrows/vxl/color_commonality_filter.cxx

```diff
@@ -0,0 +1,561 @@
+    config->get_value< unsigned >( "output_scale",
+                                   d->settings.output_scale_factor );
+  d->settings.grid_image =
+    config->get_value< bool >( "grid_image",
+                               d->settings.grid_image );
+  d->settings.grid_resolution_width =
+    config->get_value< unsigned >( "grid_resolution_width",
+                                   d->settings.grid_resolution_width );
+  d->settings.grid_resolution_height =
+    config->get_value< unsigned >( "grid_resolution_height",
+                                   d->settings.grid_resolution_height );
+
+  d->color_resolution = d->color_resolution_per_chan *
+                        d->color_resolution_per_chan *
+                        d->color_resolution_per_chan;
+
+  d->color_histogram.resize( d->color_resolution, 0 );
+  d->intensity_histogram.resize( d->intensity_resolution, 0 );
+}
+
+// ----------------------------------------------------------------------------
+bool
+color_commonality_filter
+::check_configuration( vital::config_block_sptr config ) const
+{
+  if( !is_power_of_two( d->color_resolution_per_chan ) ||
+      !is_power_of_two( d->intensity_resolution ) )
+  {
+    LOG_ERROR( logger(), "Specified resolutions must be a power of 2" );
+  }
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+kwiver::vital::image_container_sptr
+color_commonality_filter
+::filter( kwiver::vital::image_container_sptr image_data )
+{
+  // Perform Basic Validation
+  if( !image_data )
+  {
+    return kwiver::vital::image_container_sptr();
+  }
+
+  if( image_data->depth() != 1 && image_data->depth() != 3 )
+  {
+    LOG_ERROR( logger(), "Invalid number of input planes!" );
+    return kwiver::vital::image_container_sptr();
+  }
+
+  // Get input image
+  vil_image_view_base_sptr view =
+    vxl::image_container::vital_to_vxl( image_data->get_image() );
+
+  // Perform different actions based on input type
+#define HANDLE_CASE( T )                                             \
+  case T:                                                            \
+  {                                                                  \
+    using pix_t = vil_pixel_format_type_of< T >::component_type;     \
+    vil_image_view< pix_t > input = view;                            \
+    return d->compute_commonality( input );                          \
+  }                                                                  \
+
+  switch( view->pixel_format() )
+  {
+    HANDLE_CASE( VIL_PIXEL_FORMAT_BOOL );
+    HANDLE_CASE( VIL_PIXEL_FORMAT_BYTE );
+    HANDLE_CASE( VIL_PIXEL_FORMAT_SBYTE );
+    HANDLE_CASE( VIL_PIXEL_FORMAT_UINT_16 );
+    HANDLE_CASE( VIL_PIXEL_FORMAT_UINT_32 );
+    HANDLE_CASE( VIL_PIXEL_FORMAT_UINT_64 );
+#undef HANDLE_CASE
+
+    default:
+      LOG_ERROR( logger(), "Invalid type received" );
+      return image_data;
+  }
+
+  // Code not reached, prevent warning
+  return nullptr;
```

Comment:
This doesn't seem necessary, at least with any reasonable compiler. If it's really an issue, I would suggest instead shifting the complain-and-return code out of the `default:` and instead having a `default: break;`.